meta/st0x-spec.md

```diff
@@ -0,0 +1,96 @@
+# st0x Server API Specification V1
+
+## Goal
+
+Provide signed price quotes for specific trading pairs that can be used by solvers in Raindex orders. Each endpoint handles a single trading pair and returns a signed context containing quote information.
+
+## Magic Numbers
+
+- Web data metadata magic number: `0xff5dcce9b571ba42`
+- st0x API magic number: `0xff5d577062616972` ("st0x pair" in hex)
+
+## Server Advertisement
+
+Servers advertise price quote endpoints on metaboard with:
+- 8 byte magic number for st0x API
+- Remaining bytes contain UTF-8 encoded URL where the API is hosted for a specific trading pair
+
+Each trading pair requires its own advertisement. Orders reference these advertisements by their hash to save gas.
+
+## Request Format
+
+Clients (solvers) POST raw octet streams to the server URL with the following binary format:
+
+```
+[ 65 byte ECDSA signature ]  # Signs the hash of all following bytes
+[ 32 byte keccak256 hash ]   # Hash of web data from metaboard
+[ 8 byte unix timestamp ]    # Seconds since epoch
+[ 32 byte nonce ]           # Random bytes for uniqueness
+```
+
+No additional payload needed since each endpoint is pair-specific.
+
+## Server Response
+
+The server returns a SignedContextV1 structure containing:
+
+```solidity
+struct SignedContextV1 {
+    address signer;           // Account that signed the context
+    uint256[] context;        // The signed data array
+    bytes signature;          // EIP-191 compliant signature
+}
+```
+
+The `context` array contains:
+```
+[
+    uint256 ioRatio,        // Price ratio for the trading pair
+    uint256 pairHash,       // keccak256 hash of the pair name (e.g. "TSLAUSD")
```

Comment:
this would only be expressable in rainlang if the string is a compatible `Intorastring` that is being hashed

Reply:
i'm expecting yes, otherwise the rainlang author can't use the input tokens from context to lookup pairs

Reply:
the other way that we could do it is add more words to rainlang like `utf8-hash()` or something, that first convert the rain string to a utf8 string and then hash that instead